### PR TITLE
Make "Ghost" not link to 404 page

### LINKED ISSUE #6270: "Ghost" user in issue list points 404 page, but should not be clickable
* When a user creates an issue, then the name in the issue list is clickable and points to the profile (which is correct)
* When the user is deleted, then the name changes to "Ghost", but is still clickable and sends to 404 page (which is not correct)

The incorrect line in the template is:
* https://github.com/go-gitea/gitea/blob/01bd1fcd33a7af55bfbba10f391b22a631ac592c/templates/repo/issue/list.tmpl#L223

In other places ghost links are handled correctly:
* https://github.com/go-gitea/gitea/blob/01bd1fcd33a7af55bfbba10f391b22a631ac592c/templates/repo/issue/view_title.tmpl#L39-L43

I was able to reproduce the bug in:
* https://try.gitea.io/
* https://github.com/Freeyourgadget/Gadgetbridge/issues

- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes
  - [ ] No
  - [ ] Not relevant

### PR DESCRIPTION
**Pull request by oscarlofwenhamn**
Fixes #6270
* Add check to see that user id >0 before adding clickable link to user name

### CODE REVIEW

File: templates/repo/issue/list.tmpl [hunk 1]
```diff
@@ -220,7 +220,12 @@
 
 					<p class="desc">
 						{{ $timeStr := TimeSinceUnix .GetLastEventTimestamp $.Lang }}
-						{{$.i18n.Tr .GetLastEventLabel $timeStr .Poster.HomeLink .Poster.Name | Safe}}
+
+						{{if gt .Poster.ID 0}}
+							{{$.i18n.Tr "repo.issues.opened_by" $timeStr .Poster.HomeLink .Poster.Name | Safe}}
```
Comment (axifive) [inline templates/repo/issue/list.tmpl]: This changes existing behavior, why not `.GetLastEventLabel`?

File: templates/user/dashboard/issues.tmpl [hunk 2]
```diff
@@ -86,7 +86,11 @@
 							{{end}}
 
 							<p class="desc">
-								{{$.i18n.Tr "repo.issues.opened_by" $timeStr .Poster.HomeLink .Poster.Name | Safe}}
+								{{if gt .Poster.ID 0}}
+									{{$.i18n.Tr "repo.issues.opened_by" $timeStr .Poster.HomeLink .Poster.Name | Safe}}
```
Comment (axifive) [inline templates/user/dashboard/issues.tmpl]: I think `.GetLastEventLabel` should be here too

Comment (axifive) [reply templates/user/dashboard/issues.tmpl]: Снимок экрана 2019-03-22 в 13 25 18
On the closed issues tab we see "opened", but must be "closed" or "merged"

File: templates/repo/issue/list.tmpl (hunk 1 above)
Comment (oscarlofwenhamn) [reply templates/repo/issue/list.tmpl]: My bad, remediating.

Comment (oscarlofwenhamn) [reply templates/repo/issue/list.tmpl]: Done.

File: templates/user/dashboard/issues.tmpl (hunk 2 above)
Comment (axifive) [reply templates/user/dashboard/issues.tmpl]: @oscarlofwenhamn, I also found the same link here: gitea/templates/repo/issue/milestone_issues.tmpl

**Comment by axifive** [comment]
@oscarlofwenhamn, I also found the same link here: [gitea/templates/repo/issue/milestone_issues.tmpl](https://github.com/go-gitea/gitea/blob/06f00b1c59ef4af007b8fa413fe72a91e03e448e/templates/repo/issue/milestone_issues.tmpl#L206)

**Review by zeripath — APPROVED** [review]

**Review by zeripath — CHANGES_REQUESTED** [review]
Hmm... Shouldn't we just have these users link to a fake page saying that they're no longer real users. Then you don't need to have these multiple "fake" variants?

**Comment by oscarlofwenhamn** [comment]
@axifive 
> I also found the same link here: [gitea/templates/repo/issue/milestone_issues.tmpl](https://github.com/go-gitea/gitea/blob/06f00b1c59ef4af007b8fa413fe72a91e03e448e/templates/repo/issue/milestone_issues.tmpl#L206)

~~Thanks, I'll sort them out probably tomorrow.~~
Fixed in 8c67cab.

@zeripath Personally, I disagree, as I think making a link at that point leads to unnecessary jumping about on the site, which I think I myself would be frustrated with. I would rather add a hover text instead, with an explanation.

**Comment by zeripath** [comment]
It looks like you're missing one more label - issue_opened_by_fake

My only worry is that we're creating multiple copies of the same code and template.

Perhaps, it's better to use a subtemplate? I dunno, but it's certainly distasteful to have these multiple copies of the same code.

I think it's also worth looking at what the API gives. I suspect 404 is probably acceptable as a response to looking at a non existent user, but I wonder, is it always clear in our responses that the user doesn't exist?

**Comment by oscarlofwenhamn** [comment]
> It looks like you're missing one more label - issue_opened_by_fake

Thanks, I'll sort it to make the PR more complete, regardless of what we'll do with it.

> Perhaps, it's better to use a subtemplate?

That's not a bad idea at all, adding a template would mean it's all about adding a one-liner to the relevant places. I'll look into it, unless the consensus is to keep the links clickable and rather review the 404 responses.  Again, my vote is to display the "This user does not exist"-info on hover or something like that, and make Ghost non-clickable.

**Comment by zeripath** [comment]
I think you're right, making the ghost users non clickable is most right. (We probably do need a nonexistent user page in case any are missed or someone tries to craft a link to one.)

**Comment by oscarlofwenhamn** [comment]
Agreed: Say I type out an explicit link to a collaborators page in a README, and later that user is removed, we'd want to have something catching that. Though, I think that is outside the scope of this PR. Either way, I'll continue with the last label and look at making a subtemplate.

**Comment by oscarlofwenhamn** [comment]
@zeripath I've ran into an issue: I need to either pass the variable $timeStr to the new template or, if I pass the current element of .Issues and evaluate the $timeStr in the new template, I need a way to pass $.i18n into the template, as currently I can't figure out how to do both. Do you have any suggestion?

Something like 
```go
{{ range .Issues }}
...
{{ $timeStr := TimeSinceUnix .GetLastEventTimestamp $.Lang }}
{{ template "repo/issue/poster_link" {{. , $timeStr}} }}
...
```
so I can evaluate the time outside the template would be preferred, in my opinion, if possible.

File: options/locale/locale_en-US.ini [hunk 3]
```diff
@@ -770,8 +770,10 @@ issues.action_assignee = Assignee
 issues.action_assignee_no_select = No assignee
 issues.opened_by = opened %[1]s by <a href="%[2]s">%[3]s</a>
 pulls.merged_by = merged %[1]s by <a href="%[2]s">%[3]s</a>
+pulls.merged_by_fake = merged %[1]s by %[2]s
 issues.closed_by = closed %[1]s by <a href="%[2]s">%[3]s</a>
 issues.opened_by_fake = opened %[1]s by %[2]s
```
[the remark this answers is not in the document]
Comment (zeripath) [reply options/locale/locale_en-US.ini:775]: Oh cool sorry missed that

**Comment by zeripath** [comment]
I think any template to create the user link should just be as simple as possible - almost be as simple as calling a function on the user, eg. `user.Anchor()`.

If it's not clear how to do it - we can just do the fakes and think about it later.

**Review by zeripath — APPROVED** [review]

**Review by axifive — APPROVED** [review]

**Review by sapk — APPROVED** [review]

**Pull request merged**